apps/frontend/src/libs/components/input/input.tsx

```diff
@@ -33,6 +39,7 @@ const Input = <T extends FieldValues>({
 	const error = errors?.[name]?.message;
 	const hasError = Boolean(error);
 	const isRadioWithOptions = type === "radio" && options?.length;
+	const shouldShowIcon = iconName && typeof onIconClick === "function";
```

Comment:
```suggestion
	const shouldShowIcon = Boolean(iconName);
```
Icon doesn't have to be clickable

Reply:
But if I remove typeof onIconClick === "function", I get error 
  Types of property 'onClick' are incompatible.
  Type '(() => void) | undefined' is not assignable to type 'MouseEventHandler<HTMLButtonElement>'.

Reply:
Create another function within Icon component:
```
const handleIconClick = (): void => {
  onIconClick?.()
}
```
And use it

Reply:
```
	onClick?: (() => void) | undefined;

``` 
I know that onClick? means that onClick can be undefined. But without " | undefined " I get a type error.

Reply:
It's expected to be this way

Reply:
Yes, we should have `| undefined` there